File: backend/app/models/comment.py
"""Event's category table model"""
from sqlalchemy import Column, String, ForeignKey, Text, DateTime
from sqlalchemy.dialects.postgresql import UUID
from app.extensions import db
import uuid
from datetime import datetime

class Comment(db.Model):
    __tablename__ = 'comments'

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    event_id = Column(UUID(as_uuid=True), ForeignKey('events.id'), nullable=False)
    user_id = Column(UUID(as_uuid=True), ForeignKey('users.id'), nullable=False)
    content = Column(Text, nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow)

File: mobile_app/app/src/main/java/com/example/event/ui/home/Comment.java
package com.example.event.ui.home;

public class Comment {
    public String id;
    public String eventId;
    public String userId;
    public String content;
    public String createdAt;
    public String authorFirstName;
    public String authorLastName;

    public Comment(String id, String eventId, String userId, String content, String createdAt, 
                   String authorFirstName, String authorLastName) {
        this.id = id;
        this.eventId = eventId;
        this.userId = userId;
        this.content = content;
        this.createdAt = createdAt;
        this.authorFirstName = authorFirstName;
        this.authorLastName = authorLastName;
    }
}
